### PR TITLE
task dependson with it's test

### LINKED ISSUE #328: Task Dependency (V-1)
### Create a dependency section  on create task form

**Description**
- [ ] Have to create a dependency section on the task form when a task is created. 
- [ ] Have to give task-id in the dependency section.

**Requirements**

- Dependency section takes the task-id to have a dependency on another task.
- TDD is to be followed

**UI**
RDS-dependsOn

### PR DESCRIPTION
**Pull request by vinit717**
close #328

#https://github.com/Real-Dev-Squad/website-backend/pull/1070

https://github.com/Real-Dev-Squad/website-dashboard/assets/111434418/cfec5361-46c6-4418-a005-891383521c95

The issue with the task creation test
https://github.com/Real-Dev-Squad/website-dashboard/issues/359

### CODE REVIEW

File: __tests__/tasks/task-dependency.test.js
Comment (RitikJaiswal75) [inline __tests__/tasks/task-dependency.test.js:1]: what about checking the functionality whether your calls are working

File: task/script.js [hunk 1]
```diff
@@ -221,6 +223,21 @@ taskForm.onsubmit = async (e) => {
...
+  dataToBeSent.dependsOn = dataToBeSent.dependsOn.filter(
+    (dependOn) => dependOn,
+  );
+  console.log(dependsOn);
+  if (dataToBeSent.dependsOn.length !== 0) {
+    dataToBeSent.dependsOn = dataToBeSent.dependsOn[0].split(',');
+    dataToBeSent.dependsOn = dataToBeSent.dependsOn.filter(
+      (dependOn) => dependOn,
+    );
+    console.log('230', dependsOn);
```
Comment (RitikJaiswal75) [inline task/script.js:235]: is it good to keep this console log in the client machine?

File: task/script.js [hunk 2]
```diff
@@ -221,6 +223,21 @@ taskForm.onsubmit = async (e) => {
     delete dataToBeSent.links;
   }
 
+  dataToBeSent.dependsOn = dataToBeSent.dependsOn.filter(
+    (dependOn) => dependOn,
+  );
+  console.log(dependsOn);
```
Comment (RitikJaiswal75) [inline task/script.js:229]: please remove console log

Comment (vinit717) [reply task/script.js]: removed

File: task/script.js (hunk 1 above)
Comment (vinit717) [reply task/script.js]: removed

**Review by bhtibrewal — APPROVED** [review]
LGTM

**Review by bhtibrewal — APPROVED** [review]

**Pull request merged**